pkg/api/promotion.go

```diff
@@ -109,3 +109,17 @@ var (
 		}
 	}
 )
+
+// TargetName returns the target name in the registry for the promoted images
+func TargetName(registry string, config PromotionConfiguration) string {
+	if len(config.Name) > 0 {
+		if registry == QuayOpenShiftCIRepo {
+			return fmt.Sprintf("%s/%s_%s_${component}", registry, config.Namespace, config.Name)
+		}
+		return fmt.Sprintf("%s/%s/%s:${component}", registry, config.Namespace, config.Name)
+	}
+	if registry == QuayOpenShiftCIRepo {
+		return fmt.Sprintf("%s/%s_${component}_%s", registry, config.Namespace, config.Tag)
+	}
+	return fmt.Sprintf("%s/%s/${component}:%s", registry, config.Namespace, config.Tag)
+}
```

Comment:
We know in `defaults.go` which registry is used by each step, but loose that information when calling this function.  There is no need for these `if registry == …` conditionals.